src/core/settings.c

```diff
@@ -867,6 +867,106 @@ QuicSettingsLoad(
         Settings->VersionNegotiationExtEnabled = !!Value;
     }
 
+    if (!Settings->IsSet.VersionSettings) {
+        uint32_t OfferedVersionsSize = 0, AcceptableVersionsSize = 0, FullyDeployedVersionsSize = 0;
+        if (QUIC_SUCCEEDED(
```

Comment:
Why not allow for registry to overwrite just one?

Reply:
All three don't have to be updated at the same time, but all three must be present in order for version settings to work. They can all be the same list when an update is not in progress.